workshops/views.py

```diff
@@ -344,7 +344,14 @@ def admin_dashboard(request):
 
     current_events = (
         Event.objects.upcoming_events() | Event.objects.ongoing_events()
-    ).active()
+    ).active().annotate(
+        is_swc=Sum(Case(When(tags__name='SWC', then=1),
+                        default=0,
+                        output_field=IntegerField())),
+        is_dc=Sum(Case(When(tags__name='DC', then=1),
+                       default=0,
+                       output_field=IntegerField())),
+    )
```

Comment:
UPDATE: actually replacing this annotation with `prefetch_related('tags')` helps with the number of queries! I'd go this way and then merge.